### PR TITLE
sysadvisor support reclaimed core with numa binding

### PR DESCRIPTION
**Pull request by funnydreamwinz**

### CODE REVIEW

File: pkg/agent/sysadvisor/plugin/qosaware/qos_aware.go [hunk 1]
```diff
@@ -92,6 +92,15 @@ func NewQoSAwarePlugin(pluginName string, conf *config.Configuration, extraConf
 		}
 	}
 
+	if !headroomReporterExists {
```
Comment (luomingmeng) [inline pkg/agent/sysadvisor/plugin/qosaware/qos_aware.go:95]: it can also use 'resourceGetter!=nil' to determine whether it is allowed to run

Comment (funnydreamwinz) [reply pkg/agent/sysadvisor/plugin/qosaware/qos_aware.go]: fixed

File: pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go [hunk 2]
```diff
@@ -155,6 +145,87 @@ func (cs *cpuServer) ListAndWatch(_ *advisorsvc.Empty, server cpuadvisor.CPUAdvi
 	}
 }
 
+// assmble per-numa headroom
```
Comment (luomingmeng) [inline pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go:148]: typo: assemble

File: pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go [hunk 3]
```diff
@@ -155,6 +145,87 @@ func (cs *cpuServer) ListAndWatch(_ *advisorsvc.Empty, server cpuadvisor.CPUAdvi
...
+				CgroupPath: retEntry.CgroupPath,
+				CalculationResult: &advisorsvc.CalculationResult{
+					Values: general.DeepCopyMap(retEntry.Values),
+				},
+			}
+			resp.ExtraEntries = append(resp.ExtraEntries, calculationInfo)
+		}
+	}
+
+	extraNumaHeadRoom := cs.assembleHeadroom()
```
Comment (luomingmeng) [inline pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go:221]: extraNumaHeadRoom -> extraNumaHeadroom

File: pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go [hunk 4]
```diff
@@ -155,6 +145,87 @@ func (cs *cpuServer) ListAndWatch(_ *advisorsvc.Empty, server cpuadvisor.CPUAdvi
...
 
+// assmble per-numa headroom
+func (cs *cpuServer) assembleHeadroom() *advisorsvc.CalculationInfo {
+	numaAllocatable, err := cs.headroomResourceManager.GetNumaAllocatable()
+	if err != nil {
+		klog.Errorf("get numa allocatable failed: %v", err)
+		return nil
+	}
+
+	numaHeadroom := make(map[int]float64)
```
Comment (luomingmeng) [inline pkg/agent/sysadvisor/plugin/qosaware/server/cpu_server.go:156]: make(cpuadvisor.CPUNUMAHeadroom)

File: pkg/agent/sysadvisor/plugin/qosaware/server/server.go [hunk 5]
```diff
@@ -22,10 +22,13 @@ import (
 	"sync"
 	"time"
 
+	"github.com/kubewharf/katalyst-core/pkg/agent/sysadvisor/plugin/qosaware/reporter"
```
Comment (luomingmeng) [inline pkg/agent/sysadvisor/plugin/qosaware/server/server.go:25]: move to below

**Review by luomingmeng — APPROVED** [review]

**Pull request merged**